### PR TITLE
Convert params to destructured object

### PR DESCRIPTION
**Pull request by cancerberoSgx**
* https://github.com/dsherret/ts-morph/issues/598
 * tests
 * docs

### CODE REVIEW

File: src/compiler/ast/base/ParameteredNode.ts [hunk 1]
```diff
@@ -125,5 +135,18 @@ export function ParameteredNode<T extends Constructor<ParameteredNodeExtensionTy
...
+        convertParamsToDestructuredObject(formatSettings: FormatCodeSettings = {}, userPreferences: UserPreferences = {}) {
+            const params = this.getParameters();
+            if (params.length === 0) {
+                return;
+            }
+            const sourceFile = this.getSourceFile();
+            const edits = this._context.languageService.getEditsForRefactor(sourceFile.getFilePath(), {}, params[0],
+                "Convert parameters to destructured object", "Convert parameters to destructured object", {});
+            if (edits) {
+                edits.getEdits().forEach(edit => sourceFile.applyTextChanges(edit.getTextChanges()));
```
Comment (cancerberoSgx) [inline src/compiler/ast/base/ParameteredNode.ts]: this is wrong - changes could be in other files

File: src/compiler/tools/results/RefactorEditInfo.ts [hunk 2]
```diff
@@ -45,5 +45,18 @@ export class RefactorEditInfo {
         return this.compilerObject.renameLocation;
     }
 
+    /**
+     * Executes the refactor edits.
+     *
+     * WARNING: This will cause all nodes to be forgotten in the changed files.
+     * @options - Options used when applying the changes.
+     */
+    applyChanges(options?: ApplyFileTextChangesOptions) {
```
Comment (cancerberoSgx) [inline src/compiler/tools/results/RefactorEditInfo.ts]: @dsherret I thought adding this one also here makes sense, right ?

Comment (dsherret) [reply src/compiler/tools/results/RefactorEditInfo.ts]: Yes it definitely does. Thanks! Want to add a separate PR for it though?

Comment (cancerberoSgx) [reply src/compiler/tools/results/RefactorEditInfo.ts]: Sure, on it

File: src/tests/compiler/ast/base/parameteredNodeTests.ts [hunk 3]
```diff
@@ -178,4 +178,47 @@ describe(nameof(ParameteredNode), () => {
...
+    }
+}`.trim());
+
+        });
+        it("should refactor all files", () => {
+            const { sourceFile, project } = getInfoFromText<FunctionDeclaration>("export function f(a: number, b: string[]) { }", {filePath: "f.ts"});
+            const sourceFile2 = project.createSourceFile("a.ts", `import {f} = require("./f"); f(1, "b");`);
+            sourceFile.getFunctionOrThrow("f").convertParamsToDestructuredObject();
+            expect(sourceFile.getText()).to.equal(`export function f({ a, b }: { a: number; b: string[]; }) { }`);
+            expect(sourceFile2.getText()).to.equal(`import {f} = require("./f"); f(1, "b");`);
```
Comment (cancerberoSgx) [inline src/tests/compiler/ast/base/parameteredNodeTests.ts]: I'm having some trouble making it work for multiple files - this test is wrong... researching on typescript project...

**Pull request closed without merging**